sympy/core/basic.py

```diff
@@ -1284,8 +1311,26 @@ def has_free(self, *patterns):
         True
         >>> (x + y + 1).has_free(y + 1)
         True
-
         """
+        if not patterns:
+            return False
+        p0 = patterns[0]
+        if len(patterns) == 1 and iterable(p0) and not isinstance(p0, Basic):
```

Comment:
Why would this not be Basic? Is it because it needs to be sympified?

Reply:
No, because the user may fogetting to unpack an iterable. Sometimes you do (here) and sometimes you don't (`has_xfree`). There are several functions like this besides these. We could just immediately return False (GIGO).
```
            # Basic can contain iterables (though not non-Basic, ideally)
            # but don't encourage mixed passing patterns
            raise ValueError(filldedent('''
                Expecting 1 or more Basic args, not a single
                non-Basic iterable. Don't forget to unpack
                iterables: `eq.has_free(*patterns)`'''))
```